erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -324,3 +324,11 @@ def update_serial_nos_after_submit(controller, parentfield):
 						update_rejected_serial_nos = False
 						if accepted_serial_nos_updated:
 							break
+
+def update_maintenance_status():
+	last_date = frappe.utils.add_days(frappe.utils.nowdate(), -1)
+	serial_nos = frappe.db.sql('''select name from `tabSerial No` where amc_expiry_date=%s or
```

Comment:
(amc_expiry_date <= %s or warranty_expiry_date <= %s) and maintenance_status not in ('Out of Warranty', 'Out of AMC')